ui/src/AudioRecorder.js

```diff
@@ -1,44 +1,48 @@
 export class AudioRecorder {
     /**
-    * Creates an AudioRecorder that can record utterances
-    * @param {String} stores and allows access to utterance after recording
-    * @param {Object} allows access to Web Audio API
-    */
+     * Creates an AudioRecorder that can record utterances
+     * @param {String} recordingUrl - A URL that allows access to an audio file after recording.
+     * @param {Blob} blob - An audio recording file. 
+     * @param {Object} mediaRecorder - Allows access to Web Audio API.
+     */
     constructor() {
         this.recordingUrl;
+        this.blob;
         this.mediaRecorder;
     }
+
     /**
-    * Prompts user for access to Microphone using API and starts recording is access granted
-    */
+     * Prompts user for access to Microphone Component and begins recording if access is granted.
+     */
     startRecording() {
         navigator.mediaDevices.getUserMedia({ audio: true, video: false })
             .then((stream) => {
                 const mediaRecorder = new MediaRecorder(stream);
                 this.mediaRecorder = mediaRecorder;
                 mediaRecorder.start();
             }).catch(function(err) {
-                console.log(err.name, err.message); // Is this good to have or is just the alert fine?
-                alert("need access to microphone to record");
+                alert("Error: Need access to microphone to record.");
```

Comment:
also wanted to mention: last time I played with the app it never asked me for permissions, just kept saying I denied access; has that changed?

Reply:
Unfortunately no, there is no way to reprompt unless you manually remove the site from your blocked sites list in Chrome settings. I'm going to add instructions on how to do this in the error message

Reply:
for clarity, why would it end up in blocked sites though? if other sites are able to request, why can't we?

Reply:
This is a problem for all sites--with the Web Audio API on Chrome, once the user hits "Block" for any component, Chrome automatically adds it to that component's Block page. It's convenient when the user accepts (as they never have to accept again), but there does not seem to be a workaround in Chrome